### ISSUE TITLE
Feature: Qos DSCP packet marking

### ISSUE THREAD

**Issue by tiagogaspar8** [duplicati/duplicati#4734]
- [x] I have searched open and closed issues for duplicates.
- [x] I have searched the [forum](https://forum.duplicati.com) for related topics.

----------------------------------------

## Environment info
- **Duplicati version**: 2.0.6.3_beta_2021-06-17
- **Operating system**: Windows 10/11
- **Backend**: N/A

## Description
I'd like to ask for the addition of a Qos feature using DSCP, allowing one to customize it and by default being LE. This will allow better traffic scheduling allowing for a greater networking experience while backing up.

## Steps to reproduce
1. Do a Backup over the network
2. The traffic sent by duplicati isn't DSCP marked.

- **Actual result**:
No DSCP mark
- **Expected result**:
DSCP mark of LE

## Screenshots
N/A

## Debug log
Feature request, N/A

**Comment by ts678** [comment]
Could [Manage QoS Policy](https://docs.microsoft.com/en-us/windows-server/networking/technologies/qos/qos-policy-manage) give this? If giving it a path to Duplicati, don't autoupdate or the updated executable will be in an updates folder rather than in C:\Program Files. Google search isn't showing that .NET Framework is able to do DSCP directly.

**Comment by tiagogaspar8** [comment]
Yes, it could, but that would require one to manually edit windows configuration and would increase complexity in large deployments, plus, some of duplicati's traffic wouldn't probably fall in the backup category right?

**Comment by drwtsn32x** [comment]
I used Windows Group Policy to tag Duplicati traffic and it worked perfectly, when combined with the appropriate QoS rules on my router of course.

I only stopped using it because I changed my strategy to do all backups to my local NAS.  Then my NAS syncs to Cloud Storage. I set a rule on my firewall to tag that traffic from my NAS to apply a QoS rule to it.

It's not that complex to do the GP in Windows. I can provide more info if you wish. Maybe some day Duplicati will natively tag packets but it's a good workaround in the meantime.

**Comment by ackstorm23** [comment]
This is a feature that would be well received for any environment.  On my linux fileserver where I use duplicati to backup my NAS data to B2 it would also be great to tag the data as bulk so my router de-prioritized it appropriately.  

Right now I have to have firewall rules to specifically tag this traffic to do this, which is no ideal. 

I've seen other backup products implement this so I'm surprised the Duplicati devs don't understand the use-case.

**Comment by ts678** [comment]
> I'm surprised the Duplicati devs don't understand the use-case.

I'm not seeing people I would call primary developers saying that. There are hugely more feature requests than developers...
What I do see are people trying to suggest alternatives, which is pretty much what you have to use when feature isn't there.

There's also a technical question of whether it's reasonably possible to do QoS through .NET Framework. I see no clear path.
Maybe someone with more experience or motivation (maybe you?) can do more research on how one could implement this.

> I've seen other backup products implement this

If we're talking implementation, are any of those products done in C# and therefore multiplatform? That would be important.
Duplicati also uses various third-party backend libraries such as you can see in About --> Libraries. How to add this to them?

As a community supported volunteer project (low on volunteers now,) Duplicati invites people to help with features and fixes.
There is plenty that's not ideal, but Duplicati only exists and improves if people contribute their time and skills to all its needs.
